_source/_posts/2025-07-18-react-pwa.md

```diff
@@ -0,0 +1,832 @@
+}
+
+form input[type="submit"] {
+ display: block;
+ background-color: #213547;
+ box-shadow: 0 0 0 0 #213547;
+ text-transform: capitalize;
+ letter-spacing: 1px;
+ border: none;
+ color: #fff;
+ font-size: .9em;
+ text-align: center;
+ padding: 10px;
+ width: 50%;
+ margin: 15px 0 0 auto;
+ transition: background-color 250ms ease;
+ border-radius: 5px;
+}
+
+.tab {
+ max-width: 500px;
+ margin: auto;
+}
+
+form label {
+ text-align: left;
+ max-width: 100%;
+ margin-bottom: 5px;
+ font-size: 16px;
+ font-weight: 300;
+ line-height: 24px;
+}
+
+.tab-heading {
+ border-bottom: 1px solid #D9E4EEFF;
+}
+
+.tab-heading button{
+ width: 50%;
+}
+.tab-heading button:first-child{
+ text-align: right;
+}
+.tab-heading button:last-child{
+ text-align: left;
+}
+.tab-heading button:hover {
+ background: #dcdcdc;
+ border-radius: 0;
+}
+
+.tab-heading button.active{
+ background: rgba(217, 228, 238, 0.42);
+}
+.tab-heading button.active:first-child {
+ border-right: 1px solid rgba(217, 228, 238, 0.9);
+ border-bottom-left-radius: 0;
+ border-top-left-radius: 0;
+}
+
+.tab-heading button.active:last-child{
+ border-left: 1px solid rgba(217, 228, 238, 0.9);
+ border-bottom-right-radius: 0;
+ border-top-right-radius: 0;
+}
+
+.task-list {
+ list-style-type: none;
+}
+
+.task-list p.description {
+ text-align: left;
+ margin-top: 0;
+ font-size: 0.8rem;
+}
+
+.task-list li {
+ display: flex;
+ flex-direction: column;
+ border-bottom: 2px solid rgba(217, 228, 238, 0.7);
+ padding: 5px 10px;
+}
+
+.task-list li .title-card {
+ display: flex;
+}
+
+.task-list li.done .title-card *:not(.expand){
+ text-decoration: line-through;
+}
+
+.task-list li:hover {
+ background: rgba(234, 243, 252, 0.59);
+}
+
+.task-list li input[type=checkbox] {
+ margin-right: 15px;
+ cursor: pointer;
+}
+
+.task-list li p.name {
+ font-size: 1.2rem;
+}
+
+.task-list li p.expand {
+ color: #46617a;
+ font-size: 1rem;
+ margin-left: auto;
+ cursor: pointer;
+}
+```
+
+The above are helper styles. I used a tabular design for the to-do list component so CRUD can be on the same page without using modal popups. Once all the files are in place, you'll see the Todo List home page when you log in with Okta. 
+
+Once you have all the required manifest icons in your project, when you serve the app, you'll see a prompt in the browser to install it on your machine! If you don't want to create icons, use [the ones in the sample repo](https://github.com/oktadev/okta-react-pwa-example).
+
+## Authenticate with Social Login from a React PWA
+
+Great job making it this far! Along the way, we've explored how social login works with Okta and Google and how to set up a basic PWA using React, Vite, and the Vite PWA plugin. As a bonus, we now have a handy little to-do list app to help keep our day on track!
+
+Of course, a production-ready application would involve more advanced service worker configurations and a proper database setup, but our current implementation is adequate for an introduction.
```

Comment:
Remove filler like “Of course” and avoid words like would.

You can write this line - 
A production-ready application requires more advanced service-worker configuration and a real database, but the current implementation is apt for an introduction.

Reply:
That's so formal. I disagree but in principle, I do agree that we should avoid "of course" as it's not inclusive